### PR TITLE
Add klusterlet degraded status

### PR DESCRIPTION
**Pull request by qiujian16**

### CODE REVIEW

File: pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+			name:       "No cluster name secret",
+			object:     []runtime.Object{newSecret(bootstrapHubKubeConfigSecret, "test"), newSecret(hubKubeConfigSecret, "test")},
+			klusterlet: newKlusterlet("testklusterlet", "test", ""),
+			expectedConditions: []opratorapiv1.StatusCondition{
+				testinghelper.NamedCondition(klusterletRegistrationDegraded, "ClusterNameMissing", metav1.ConditionTrue),
+				testinghelper.NamedCondition(klusterletWorKDegraded, "ClusterNameMissing", metav1.ConditionTrue),
+			},
+		},
+		{
+			name:       "No kubecoonfig secret",
```
Comment (pmorie) [inline pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller_test.go]: `kubeconfig`

File: pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+				helpers.UpdateKlusterletConditionFn(workDegradedCondition),
+			)
+			return err
+		}
+	}
+
+	// If hub kubeconfig does not exist, return err.
+	if hubConfigSecret.Data["kubeconfig"] == nil {
+		registrationDegradedCondition.Message = fmt.Sprintf(
+			"Failed to get cluster name from `kubectl get secret -n %q %q -ojsonpath='{.data.kubeconfig}`.  This is set by the klusterlet registration deployment, but the CSR must be approved by the cluster-admin on the hub.", hubConfigSecret.Namespace, hubConfigSecret.Name)
```
Comment (pmorie) [inline pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go]: Failed to get kubeconfig

File: pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+	)
+	return nil
+}
+
+func (k *klusterletStatusController) queueKeyFunc(obj runtime.Object) string {
+	accessor, _ := meta.Accessor(obj)
+	namespace := accessor.GetNamespace()
+	name := accessor.GetName()
+
+	// return empty key if secret ot deployment is not interested
```
Comment (pmorie) [inline pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go]: `secret or deployment is not interesting`

File: pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller_test.go [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kubeinformers "k8s.io/client-go/informers"
+	fakekube "k8s.io/client-go/kubernetes/fake"
+	clienttesting "k8s.io/client-go/testing"
+
+	fakeoperatorclient "github.com/open-cluster-management/api/client/operator/clientset/versioned/fake"
+	operatorinformers "github.com/open-cluster-management/api/client/operator/informers/externalversions"
+	opratorapiv1 "github.com/open-cluster-management/api/operator/v1"
```
Comment (pmorie) [inline pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller_test.go]: `operatorapiv1` or `operatorv1`

**Review by pmorie — COMMENTED** [review]
Looking good, just found one condition w/ incorrect text

**Comment by qiujian16** [comment]
updated and add an integration test.

**Comment by deads2k** [comment]
very nice.

/lgtm
/approve

**Pull request merged**